docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,236 @@
+
+This documentation focuses on how to test Apostrophe modules, not Apostrophe itself using integration testing.
+
+Modules should be stand-alone npm packages.
+
+We will use a fictitious module named `article` to illustrate how to write tests for modules.
+
+## Requirements
+
+- A running MongoDB server
+
+## Setup
+
+This setup assumes you will use the following packages
+
+- [apostrophe](https://www.npmjs.com/package/apostrophe): *ApostropheCMS is a full-featured, open source CMS built with Node.js that seeks to empower organizations by combining in-context editing and headless architecture in a full-stack JS environment*
+- [eslint](https://www.npmjs.com/package/eslint): *ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code*
+- [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe): *An ESLint configuration for ApostropheCMS core and officials modules*
+- [mocha](https://www.npmjs.com/package/mocha): *Simple, flexible, fun JavaScript test framework for Node.js & The Browser*
+```sh
+npm install --save-dev apostrophe eslint eslint-config-apostrophe mocha
+```
+
+You will need to add some new scripts to your `package.json`.
+
+```json
+{
+  "name": "article",
+  "version": "1.0.0",
+  "description": "A fictitious Apostrophe module",
+  "main": "index.js",
+  "scripts": {
+    "lint": "npm run eslint",
+    "eslint": "eslint .",
+    "test": "npm run lint && mocha"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC"
+}
+```
+
+Add the following to `.eslintrc.json` to tell eslint to use ApostropheCMS ESLint confiuration.
+
+```json
+{
+  "extends": [ "apostrophe" ]
+}
+```
+
+When you call `npm test`, it will lint your files according to [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe) rules and run the tests using [mocha](https://www.npmjs.com/package/mocha).
+
+### Test folder
+
+You will need a `test/package.json` file referencing the repository URL of your module. Please replace `article` & `%article-repository-url%` with your module informations.
+
+e.g. for the module [@apostrophecms/login-hcaptcha](https://github.com/apostrophecms/login-hcaptcha) we use 
+
+> `"@apostrophecms/login-hcaptcha": "git://github.com/apostrophecms/login-hcaptcha.git"`
+> 
+
+```json
+{
+  "/**": "This package.json file is not actually installed.",
+  " * ": "Apostrophe requires that all npm modules to be loaded by moog",
+  " */": "exist in package.json at project level, which for a test is here",
+  "dependencies": {
+    "apostrophe": "^3.26.0",
+    "article": "%article-repository-url%"
+  }
+}
+```
+
+You don’t want to commit test files artifacts to git, please add the following to your `.gitignore` file
+
+```gitignore
+# Test files
+/test/apos-build
+/test/public
+/test/locales
+```
+
+## Writing tests
+
+We start with a single test file `test/index.js`.
+
+As your tests grows, you can break it down into multiple files based on your liking.
```

Comment:
"As your test grows, ..."
or
"As your tests grow, you can break them down..."

Reply:
Sorry, my mistake.
Grammatically the first one doesn't make exact sense. Use:
"As your tests grow, you can break them down ..."